scala/private/rule_impls.bzl

```diff
@@ -498,8 +510,24 @@ def _compile_or_empty(
             ijars = ijars,
             java_jar = java_jar,
             source_jars = source_jars,
+            merged_provider = merged_provider
         )
 
+def _create_scala_compilation_provider(ctx, ijar, deps_providers):
+    exports = []
+    if hasattr(ctx.attr, "exports"):
+        exports = [dep[JavaInfo] for dep in ctx.attr.exports]
+    runtime_deps = []
+    if hasattr(ctx.attr, "runtime_deps"):
+        runtime_deps = [dep[JavaInfo] for dep in ctx.attr.runtime_deps]
+    return JavaInfo(
```

Comment:
I just realized that this doesn’t pass the source_jar.
Any chance you can add that? If not I’ll add it in a separate PR

Reply:
I'm not sure which is the source jar here. Is it `_pack_source_jars(ctx)`? Note that `JavaInfo()` takes only one source jar, so probably there needs to be some more packing before.